SUILesson4/SUILesson4/ContentView.swift

```diff
@@ -6,16 +6,203 @@
 //
 
 import SwiftUI
+import AVFoundation
 
+/// Контент представления с плеером
 struct ContentView: View {
+
+    // MARK: Private constants
+
+    private enum Constants {
+        static let rowDownImageName = "arrow.down.circle"
+        static let trackDownloadedName = "Запись успешно загружена"
+        static let rowUpImageName = "arrow.up.forward.app"
+        static let trackDurationName = "03:58"
+        static let trackStartTimeName = "0:00"
+        static let rowLeftImageName = "arrow.left.to.line"
+        static let playImageName = "play.fill"
+        static let pauseImageName = "pause.fill"
+        static let stopImageName = "stop.fill"
+        static let rowRightImageName = "arrow.right.to.line"
+        static let zeroName = "0"
+        static let hundredName = "100"
+    }
+
+    // MARK: - public properties
+
+    @ObservedObject var viewModel = PlayerViewModel()
+
     var body: some View {
-        VStack {
-            Image(systemName: "globe")
-                .imageScale(.large)
-                .foregroundColor(.accentColor)
-            Text("Hello, world!")
+        ZStack {
+            RoundedRectangle(cornerRadius: 1)
+            VStack {
+                HStack {
+                    makeDownLoadButton
+                    makeSongPosterImage
+                    makeShareSongButton
+                }
+                HStack {
+                    makeInfoAboutSongSection
+                }
+                .frame(height: 10)
+                .padding()
+
+                makeSongDuratonSlider
+
+                HStack {
+                    makePlayPreviusSongButton
+                    makePlaySongButton
+                    makePauseSongButton
+                    makeStopSongButton
+                    makePlayNextSongButton
+                }
+                makeVolumeSlider
+            }.background(Color.black)
+        }
+    }
+
+    // MARK: Private properties
+
+    @State private var progress: Double = 0
+    @State private var volume: Float = 0
+    @State private var isDownloadActionSheertShown = false
+    @State private var isEditing = false
+
+    private var makeDownLoadButton: some View {
+        Button {
+            self.isDownloadActionSheertShown = true
+        } label: {
+            Image(systemName: Constants.rowDownImageName)
+                .scaleEffect(x: 2, y: 2)
+                .foregroundColor(.green)
+        }.confirmationDialog(Constants.trackDownloadedName, isPresented: $isDownloadActionSheertShown, titleVisibility: .visible) {}
+    }
+
+    var makeSongPosterImage: some View  {
+        Image(self.viewModel.musicAlbum[viewModel.currentSongIndex ?? 0].imageName)
+            .resizable()
+            .scaledToFit()
+            .padding()
+            .frame(width: 250)
+    }
+
+    private var makeShareSongButton: some View {
```

Comment:
```suggestion
    private var makeShareSongButtonView: some View {
```
в остальных тоже добавь